core/domain/rte_component_registry_test.py

```diff
@@ -174,6 +169,13 @@ def test_rte_components_are_valid(self):
                 % hyphenated_component_id)
             main_html_file = os.path.join(
                 directives_dir, '%s.directive.html' % hyphenated_component_id)
+            if hyphenated_component_id == 'svgdiagram':
+                main_ts_file = os.path.join(
+                    directives_dir, 'oppia-noninteractive-%s.component.ts'
+                    % hyphenated_component_id)
+                main_html_file = os.path.join(
+                    directives_dir, '%s.component.html'
+                    % hyphenated_component_id)
```

Comment:
Why is this part needed?

Reply:
Except svgdiagram all are directive files whereas svgdiagram is a component file, so this additional condition is required.

Reply:
I will add a similar comment in the code.

Reply:
With the Angular migration will the other also become components? Then maybe use `TODO` here that this will be fixed after the Angular migratin.

Reply:
Added the todo #9762 .Done.

Reply:
Thx